packages/mui-base/src/ButtonUnstyled/ButtonUnstyled.types.ts

```diff
@@ -62,7 +62,7 @@ export type ButtonUnstyledOwnerState = ButtonUnstyledOwnProps & {
 export type ButtonUnstyledRootSlotProps = Simplify<
   UseButtonRootSlotProps & {
     ownerState: ButtonUnstyledOwnerState;
-    className: string;
+    className?: string;
```

Comment:
Why did we changed it? We are indeed always receiving the `className`.

Reply:
I made the `useSlotProps` return `undefined` if no classes are present. Having its return type depend on whether any of the inputs contain the `className` prop would significantly raise its complexity. I don't think it's worth it. Plus, we're going to allow developers to disable the built-in classes at some point, so we'd have to revert this change.